packages/client/tests/functional/chunking-query/tests.ts

```diff
@@ -0,0 +1,265 @@
+          where: {
+            id: {
+              in: ids,
+            },
+          },
+        })
+        return tags
+      }
+
+      async function clean() {
+        const cleanPrismaPromises = [prisma.tagsOnPosts.deleteMany(), prisma.post.deleteMany(), prisma.tag.deleteMany()]
+        for (const promise of cleanPrismaPromises) {
+          await promise
+        }
+      }
+
+      afterEach(async () => {
+        await clean()
+      }, 80_000)
+
+      test('should succeed when "in" has 32767 ids', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32767 records', async () => {
+        const n = 32767
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32768 ids', async () => {
+        const n = 32768
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32768 records', async () => {
+        const n = 32768
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32769 ids', async () => {
+        const n = 32769
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32769 records', async () => {
+        const n = 32769
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 65537 ids', async () => {
+        const n = 65537
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 65537 records', async () => {
+        const n = 65537
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32766 ids and a "skip" filter', async () => {
+        const n = 32766
+        const ids = await createTags(n)
+
+        const tags = await prisma.tag.findMany({
+          where: {
+            id: { in: ids },
+          },
+          skip: 1,
+        })
+
+        expect(tags.length).toBe(n - 1)
+      })
+
+      test('should succeed when the explicit number of params is 32767 or less', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsParams(ids)
+
+        expect(tags.length).toBe(n)
+      })
```

Comment:
Unrelated to this PR:
1. All these tests would benefit from an explanation _why_ they succeed. Either because the number of params is fine for one query, or because we detect the too high number and split into multiple queries, right?
2. Additionally it would be really nice if we would really confirm the number of queries that are executed. That would also "document" what the chunking is doing.

Reply:
https://github.com/prisma/prisma/issues/21827